### ISSUE TITLE
rosdep fails with rolling

### ISSUE THREAD

**Issue by roncapat** [ros/rosdistro#40236]
When issuing `ROS_DISTRO=rolling rosdep install -r -s` rosdep fails to find many packages. The result is that my `rolling` CI fails with messages like:
```
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
package_x: No definition of [geometry_msgs] for OS version [jammy]
package_x: No definition of [image_common] for OS version [jammy]
package_x: No definition of [image_transport] for OS version [jammy]
package_x: No definition of [yaml_cpp_vendor] for OS version [jammy]
package_x: No definition of [pcl_conversions] for OS version [jammy]
package_x: No definition of [pcl_ros] for OS version [jammy]
package_x: No definition of [point_cloud_transport] for OS version [jammy]
package_x: No definition of [rosidl_default_generators] for OS version [jammy]
package_x: No definition of [sensor_msgs] for OS version [jammy]
package_x: No definition of [tf2_geometry_msgs] for OS version [jammy]
Continuing to install resolvable dependencies...
```

I'm using up-to-date `rolling-desktop-full-jammy` and `rolling-desktop` images.

Am I right to suspect this has some kind of linking with the preparation of rolling to switch on top of `Ubuntu 24.04 Noble`?

**Comment by rcp1-beg** [comment]
Same here, e.g., rosdep fails for actuator_msgs:
`rosdep resolve actuator_msgs --rosdistro=rolling
WARNING: given --rosdistro rolling but ROS_DISTRO is "humble". Ignoring environment.

ERROR: No definition of [actuator_msgs] for OS version [jammy]

No definition of [actuator_msgs] for OS version [jammy]
	rosdep key : actuator_msgs
	OS name    : ubuntu
	OS version : jammy
	Data:
_is_ros: true
		debian:
		  bookworm:
		    apt:
		      packages:
		      - ros-rolling-actuator-msgs
		osx:
		  homebrew:
		    packages:
		    - ros/rolling/actuator_msgs
		rhel:
		  '9':
		    dnf:
		      packages:
		      - ros-rolling-actuator-msgs
		ubuntu:
		  noble:
		    apt:
		      packages:
		      - ros-rolling-actuator-msgs
`

**Comment by mjcarroll** [comment]
Yes, things are in flux with the transition to 24.04.  Once migration is finished, things should quiet down a bit.

**Comment by mjcarroll** [comment]
@roncapat and @rcp1-beg can you check and see if things are "back to normal" here?

**Comment by roncapat** [comment]
@mjcarroll I still have CIs based on top of `osrf/ros-rolling-desktop` failing because `hardware_interface` package is not found.

**Comment by roncapat** [comment]
According to https://hub.docker.com/r/osrf/ros/tags?page=&page_size=&ordering=&name=rolling the rolling images have been updated 1 month ago, not sure if it's related to this.

**Comment by rcp1-beg** [comment]
> @roncapat and @rcp1-beg can you check and see if things are "back to normal" here?

You mean by using rolling for jammy / 22.04 *or* for noble / 24.04?

**Comment by clalancette** [comment]
Given that we have now switched rolling over to Noble completely, I'm going to close this out.
